src/shims/unix/sync.rs

```diff
@@ -16,6 +16,11 @@ use crate::*;
 /// in `pthread_mutexattr_settype` function.
 const PTHREAD_MUTEX_NORMAL_FLAG: i32 = 0x8000000;
 
+#[cfg(not(target_os = "macos"))]
+const PTHREAD_MUTEX_ID: u64 = 0;
+#[cfg(target_os = "macos")]
+const PTHREAD_MUTEX_ID: u64 = 4;
```

Comment:
I'm afraid this will not work. You are testing the host OS Miri is running on, but you need to test the target OS that the interpreted program is running for. These are entirely independent.

You'll have to define a function, `mutex_id_offset`, that takes as argument an `ecx: &MiriInterpCx<'mir, 'tcx>` and checks the `ecx.tcx.sess.target.os` to determine the offset.